brain-marks/Categories/CategoryListViewModel.swift

```diff
@@ -10,32 +10,69 @@ import SwiftUI
 final class CategoryListViewModel: ObservableObject {
     
     @Published var categories = [AWSCategory]()
+    @Published var categorySheetState: CategoryState = .new
+    @Published var editCategory: AWSCategory?
+    @Published var indexSetToDelete: IndexSet?
+    @Published var showAddURLView = false
+    @Published var showInfoSheet = false
+    @Published var showingCategorySheet = false
+    @Published var showingDeleteActionSheet = false
+    @Published var lastEditedCategoryID = ""
+
+    var dataStoreManager: DataStoreManager = AmplifyDataStoreManger.shared
     
-    var lastEditedCategoryID = ""
-    
-    func getCategories() {
+    func getCategories(completion: (() -> Void)? = nil) {
```

Comment:
The optional completion helps for testing async code. I'd always like to avoid altering the app code for tests but in this case I think this may be the only way to handle async testing. If you have any other ideas or suggestions I'm all ears :)